benchexec/tools/proton.py

```diff
@@ -0,0 +1,70 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from benchexec.tools.sv_benchmarks_util import get_data_model_from_task, ILP32, LP64
+import benchexec.tools.template
+import benchexec.result as result
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    PROTON
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("proton")
+
+    def version(self, executable):
+        return self._version_from_tool(executable, use_stderr=True)
+
+    def name(self):
+        return "PROTON"
+
+    def cmdline(self, executable, options, task, rlimits):
+        if task.property_file:
+            options = options + ["--propertyFile", task.property_file]
+
+        data_model_param = get_data_model_from_task(task, {ILP32: "32", LP64: "64"})
+        if data_model_param and data_model_param not in options:
+            options += [data_model_param]
+
+        self.options = options
+
+        return [executable] + options + list(task.input_files_or_identifier)
+
+    def determine_result(self, run):
+        output = run.output
+
+        if run.exit_code.value in [0, 10]:
+            status = result.RESULT_ERROR
+            if len(output) > 0:
+                # SV-COMP mode
+                result_str = output[-1].strip()
+
+                if result_str == "TRUE":
+                    status = result.RESULT_TRUE_PROP
+                elif "FALSE" in result_str:
+                    if result_str == "FALSE(termination)":
+                        status = result.RESULT_FALSE_TERMINATION
+                    else:
+                        status = result.RESULT_FALSE_REACH
+                elif "UNKNOWN" in output:
+                    status = result.RESULT_UNKNOWN
+
+        elif run.exit_code.value == 64 and "Usage error!" in output:
+            status = "INVALID ARGUMENTS"
+
+        elif run.exit_code.value == 6 and "Out of memory" in output:
```

Comment:
These conditions with `in output` will check if there is a line in the output that contains exactly this string (and nothing else). Is this intended?

Especially for `UNKNOWN` I am wondering because for `FALSE` and `TRUE` only the last line is checked, and for `FALSE` even with a substring match.

Reply:
I would like to get confirmation from you that this inconsistency between e.g. `TRUE` and `UNKNOWN` is indeed intendend.

Reply:
Hm, with the current code there is still inconsistency with regards wo `... in result_str`, `result_str == ...` and `... in output`, all three of which have different semantics. Is this intended?

Did you [test this](https://github.com/sosy-lab/benchexec/blob/main/doc/tool-integration.md#testing-the-tool-integration)?

Note that for future maintainers of this logic it might be useful to have example output lines of the tool here, if it is not as trivial as just a line consisting only of the specific string.